src/systems/shadow-system.js

```diff
@@ -0,0 +1,73 @@
+import resizeShadowCameraFrustum from "../utils/resizeShadowCameraFrustum";
+
+function traverseAnimationTargets(rootObject, animations, callback) {
+  if (animations && animations.length > 0) {
+    for (const animation of animations) {
+      for (const track of animation.tracks) {
+        const { nodeName } = THREE.PropertyBinding.parseTrackName(track.name);
+        let animatedNode = rootObject.getObjectByProperty("uuid", nodeName);
+
+        if (!animatedNode) {
+          animatedNode = rootObject.getObjectByName(nodeName);
+        }
+
+        if (animatedNode) {
+          callback(animatedNode);
+        }
+      }
+    }
+  }
+}
+
+export class ShadowSystem {
+  constructor(sceneEl) {
+    this.needsUpdate = false;
+    this.dynamicShadowsEnabled = window.APP.store.state.preferences.enableDynamicShadows;
+    this.sceneEl = sceneEl;
+    this.onEnvironmentSceneLoaded = this.onEnvironmentSceneLoaded.bind(this);
+    this.sceneEl.addEventListener("environment-scene-loaded", this.onEnvironmentSceneLoaded);
+  }
+
+  onEnvironmentSceneLoaded({ detail: scene }) {
+    this.environmentObject3D = scene;
+    this.needsUpdate = true;
+    this.sceneEl.renderer.shadowMap.autoUpdate = this.dynamicShadowsEnabled;
+  }
+
+  tick() {
+    if (!this.needsUpdate) {
+      return;
+    }
+
+    if (window.APP && window.APP.quality === "low") {
+      return;
+    }
+
+    const environmentObject3D = this.environmentObject3D;
+
+    if (!environmentObject3D) {
+      return;
+    }
+
+    if (!this.dynamicShadowsEnabled) {
+      // Recompute the shadowmap when the environment changes.
```

Comment:
Is this comment correct? Seems like this block of code is just disabling shadows on animated objects. Assuming so, this code is pretty self describing, so I would just ditch the comment.